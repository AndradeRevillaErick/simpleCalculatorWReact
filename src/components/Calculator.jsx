import { useCalculator } from "../hooks/useCalculator"

export const Calculator = () => {

    const { onHandleOperation, onHandleClear, onHandleDelete, onHandleResult, operation, result } = useCalculator();

    return(
        <>
            <h1>{ operation }</h1>
            <h4 >{ result }</h4>
            <button onClick={ onHandleClear }>AC</button>
            <button onClick={ onHandleDelete }>DE</button>
            <button onClick={ () => onHandleOperation('.') }>.</button>
            <button onClick={ () => onHandleOperation('/') }>/</button>
            <button onClick={ () => onHandleOperation('*') }>*</button>
            <button onClick={ () => onHandleOperation('-') }>-</button>
            <button onClick={ () => onHandleOperation('+') }>+</button>
            <button onClick={ () => onHandleOperation(9) } >9</button>
            <button onClick={ () => onHandleOperation(8) } >8</button>
            <button onClick={ () => onHandleOperation(7) } >7</button>
            <button onClick={ () => onHandleOperation(6) } >6</button>
            <button onClick={ () => onHandleOperation(5) } >5</button>
            <button onClick={ () => onHandleOperation(4) } >4</button>
            <button onClick={ () => onHandleOperation(3) } >3</button>
            <button onClick={ () => onHandleOperation(2) } >2</button>
            <button onClick={ () => onHandleOperation(1) } >1</button>
            <button onClick={ () => onHandleOperation(0) } >0</button>
            <button onClick={ () => onHandleOperation('00') } >00</button>
            <button onClick={ onHandleResult } >=</button>
        </>
    )
}