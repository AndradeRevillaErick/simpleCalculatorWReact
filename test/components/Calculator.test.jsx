import { fireEvent, render, screen } from "@testing-library/react";
import { Calculator } from "../../src/components/Calculator";


describe('Tests in Calculator component', () => { 
    
    test('Should match with the snapshoot', () => {
        const { container } = render( <Calculator/> );

        expect( container ).toMatchSnapshot();
    });

    test('Should change the result from the h4 result', () => {

    });

    test('Should change the content from the h1 screen', () => {
        render( <Calculator/> );
        fireEvent.click( screen.getAllByRole('button') );

        expect( screen.getByText('Syntax Error'));
    }); 

    test('Shoul show Syntax Error message if you click = with an empty number', () => {
        render( <Calculator/> );
        fireEvent.click( screen.getByText('=') );

        expect( screen.getByText('Syntax Error') );
    });
 })