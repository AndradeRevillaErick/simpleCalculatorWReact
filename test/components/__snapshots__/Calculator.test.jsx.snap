// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Calculator component Should match with the snapshoot 1`] = `
<div>
  <h1 />
  <h4>
    0
  </h4>
  <button>
    AC
  </button>
  <button>
    DE
  </button>
  <button>
    .
  </button>
  <button>
    /
  </button>
  <button>
    *
  </button>
  <button>
    -
  </button>
  <button>
    +
  </button>
  <button>
    9
  </button>
  <button>
    8
  </button>
  <button>
    7
  </button>
  <button>
    6
  </button>
  <button>
    5
  </button>
  <button>
    4
  </button>
  <button>
    3
  </button>
  <button>
    2
  </button>
  <button>
    1
  </button>
  <button>
    0
  </button>
  <button>
    00
  </button>
  <button>
    =
  </button>
</div>
`;
